packages-backend/express/src/auth.ts

```diff
@@ -104,15 +104,15 @@ const getConfiguredDefaultIssuer = (options: TSessionMiddlewareOptions) => {
 // Construct the audience from the given option + the request path.
 // If the request path is `/`, do not append it to the audience, otherwise
 // the token validation might fail because of mismatching audiences.
-const getConfiguredAudience = (
+export const getConfiguredAudience = (
   options: TSessionMiddlewareOptions,
   requestPath: string
 ) => {
   const url = new URL(options.audience);
   if (requestPath !== '/') {
-    url.pathname = requestPath;
+    return `${url.origin}${requestPath}${url.search}`;
```

Comment:
The `requestPath` already contains the search query

```suggestion
    return `${url.origin}${requestPath}`;
```